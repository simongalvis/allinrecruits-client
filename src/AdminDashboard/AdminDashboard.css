
li{
    list-style-type: none;
    padding: 0;
    font-size: 33px;
    height: 95px;
    border: 1px solid black;
    margin: 3%;
    border-radius: 3px;
    max-width: 500px;
}
ul{
    display: flex;
    flex-direction: column;
    padding: 0;
}
a{
    color:black;
    text-decoration: none;
}
.AdminDashboard{
    height:100%;
}
.job-tile:hover{
background-color: rgb(241, 90, 41, .5);
}
.job-tile{
    display: flex;
    justify-content: center;
    align-items: center;
}
#subject-btn-title{
    font-size: 27px;
    
}

@media only screen and (min-width: 777px) {
    .job-tile{
        width:500px;
    }
    .tiles-container{
        display: flex;
        justify-content: center;
        align-items: center;
    } 
  }