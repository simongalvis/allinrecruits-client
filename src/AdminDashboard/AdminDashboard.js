import React from 'react';
import './AdminDashboard.css'
import NavBar from '../NavBar/NavBar'
import { Link } from 'react-router-dom';
import ApiContext from '../ApiContext';

class AdminDashboard extends React.Component{
static contextType = ApiContext;





    render(){

        const subjects = ['Spanish','French','Robotics','Art']


        return(
            <div className="AdminDashboard">
                <NavBar/>
                <h1>AdminDashboard</h1>
                <ul id="job-position-tiles">
                {
                    subjects.map(subject =>(
                
                    <Link to="applicant-list" key={subject}>
                        <li className="job-tile"   onClick={ e => this.context.selectSubject(subject)}><b>{subject}</b></li>
                    </Link>
                
                    ))     
                }
                </ul>
            </div>
        )
    }
}
export default AdminDashboard;