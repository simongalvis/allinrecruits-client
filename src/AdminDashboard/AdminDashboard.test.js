import React from 'react'
import ReactDOM from 'react-dom'
import AdminDashboard from './AdminDashboard'
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom'

describe(`AdminDashboard component`, () =>{
    it(`renders without crashing`, () =>{
        const div = document.createElement('div');

        ReactDOM.render(<BrowserRouter><AdminDashboard/></BrowserRouter>, div)

        ReactDOM.unmountComponentAtNode(div);
    })
    it(`Renders UI as expected`, () =>{
        const tree = renderer
            .create(<BrowserRouter><AdminDashboard/></BrowserRouter>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})