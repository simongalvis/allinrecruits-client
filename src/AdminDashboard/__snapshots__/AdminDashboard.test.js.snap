// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminDashboard component Renders UI as expected 1`] = `
<div
  className="AdminDashboard"
>
  <div
    className="NavBar"
  >
    <header>
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="all-in-education-logo"
          id="main-logo"
          src="all-in-logo.png"
        />
      </a>
      <a
        href="/admin-login"
      >
        <button
          id="admin-login-button"
        >
          Admin Login
        </button>
      </a>
    </header>
  </div>
  <h1>
    Subjects:
  </h1>
  <ul
    id="job-position-tiles"
  >
    <a
      href="/applicant-list"
      onClick={[Function]}
    >
      <li
        className="job-tile"
        onClick={[Function]}
      >
        <b>
          Spanish
        </b>
      </li>
    </a>
    <a
      href="/applicant-list"
      onClick={[Function]}
    >
      <li
        className="job-tile"
        onClick={[Function]}
      >
        <b>
          French
        </b>
      </li>
    </a>
    <a
      href="/applicant-list"
      onClick={[Function]}
    >
      <li
        className="job-tile"
        onClick={[Function]}
      >
        <b>
          Robotics
        </b>
      </li>
    </a>
    <a
      href="/applicant-list"
      onClick={[Function]}
    >
      <li
        className="job-tile"
        onClick={[Function]}
      >
        <b>
          Art
        </b>
      </li>
    </a>
  </ul>
</div>
`;
